app/src/main/java/com/example/trainingintive/dog_images_feature/presentation/AlbumViewModel.kt

```diff
@@ -26,27 +30,54 @@ class AlbumViewModel @Inject constructor(
     }
 
     private fun getAllImages() {
-        repository.getAllImageUrls()
-            .subscribeOn(schedulers.io())
-            .observeOn(schedulers.ui())
-            .subscribe(
-                { _imageUrls.value = it },
-                {
-                    val errorMessageId = ErrorMessageId.getId(it as Exception)
-                    navigator.sendEvent(
-                        MainScreenEvent.Error(errorMessageId)
-                    )
-                }
-            )
+        disposables +=
+            repository.getAllImageUrls()
+                .subscribeOn(schedulers.io())
+                .observeOn(schedulers.ui())
+                .subscribe(
+                    { _imageUrls.value = it.sortedBy { it.position } },
+                    {
+                        val errorMessageId = ErrorMessageId.getId(it as Exception)
+                        navigator.sendEvent(
+                            MainScreenEvent.Error(errorMessageId)
+                        )
+                    }
+                )
     }
 
-    fun changeImagePosition(from: Int, to: Int) {}
-
     fun removeImage(index: Int) {
         val imageUrl = _imageUrls.value!![index]
-        repository.remove(imageUrl)
-            .subscribeOn(schedulers.io())
-            .observeOn(schedulers.ui())
-            .subscribe()
+        disposables +=
+            repository.remove(imageUrl)
+                .subscribeOn(schedulers.io())
+                .observeOn(schedulers.ui())
+                .subscribe()
+    }
+
+    fun changeImagePositionAndUpdateDb(from: Int, to: Int) {
+        if (shouldBeChanged()) {
+            lastChangePositionTimestamp = System.currentTimeMillis()
```

Comment:
Try to find another way for this, maybe some online articles would be helpful because I'm pretty sure that they can be achieved in a simpler way. Even if you don't and you have to do it, VW is not the place for it.

Reply:
what do you think about this https://stackoverflow.com/questions/35920584/android-how-to-catch-drop-action-of-itemtouchhelper-which-is-used-with-recycle